### ISSUE TITLE
After Update to PRTG  Version 21.2.67.1562 (Stable) get-probe does not give anything back

### ISSUE THREAD

**Issue by carsten0407** [lordmilko/PrtgAPI#215]
**What's going on?**

Hi Lord,

after we Update to PRTG  Version 21.2.67.1562 (Stable)
the command --> get-probe does not give anything back

get-probe -name "*NAME*" ist functionally

Do you know anythibg about that?

Thanks

Carsten

**Comment by lordmilko** [comment]
Hi @carsten0407,

When I run `Get-Probe` against PRTG 21.2.67.1562 I get an error; this appears to be caused by the new "PRTG Core Server" autonomous device object. This object is a device, however because it is under the root node PRTG is erroneously returning it from API requests for probes as well. I will open a case regarding this with Paessler support.

**Comment by AlphaJosh** [comment]
Workaround:

$a = New-SearchFilter ProbeStatus NotEquals "" -Illegal
Get-Probe -Verbose -Filter $a

**Comment by lordmilko** [comment]
Hi @AlphaJosh,

I've been looking into this and have found that the true workaround that is required is to have a permanent filter on probes for `Type` equals `Probe`; in the meantime users can implement this manually by doing `flt type eq probenode|get-probe` or really any other filter which causes only `Probe` objects to be returned and not `AutonomousDevice` objects (as you have done)

As this has the potential to affect a large number of users I am currently working to release PrtgAPI 0.9.15 ASAP which will include a proper fix for this issue. I hope to release this within the next week or so

Regards,
lordmilko

**Comment by lordmilko** [comment]
Hi @carsten0407 / @AlphaJosh ,

Please be advised PrtgAPI 0.9.15 has now been released which includes a fix for this issue

To update PrtgAPI, please run `Update-Module PrtgAPI` and reopen PowerShell

Regards,
lordmilko

**Comment by carsten0407** [comment]
Hi Lordmilko,

it works again, but I didn't expect anything else

Good work,


thanx ;-)

**Comment by carsten0407** [comment]
Hi Lordmilko,

it is strange, but every probe I added, after the last PRTG Update,
I did not find it with get-probe –name Probename

If I grab after get-sensor –tags probeTags  see the probe but nothing with get-probe –name Probename

In the gui, the probes are visible.


I already updated to the current PRTGAPI-Version  PrtgAPI 0.9.15

Thanx

Carsten




Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: lordmilko ***@***.***>
Gesendet: Freitag, 7. Mai 2021 07:39
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Closed #215<https://github.com/lordmilko/PrtgAPI/issues/215> via 2b50641<https://github.com/lordmilko/PrtgAPI/commit/2b50641feb8541ae1b332fbbff9ebd3e1335f5f9>.

**Comment by lordmilko** [comment]
Hi @carsten0407,

Can youu please do

```powershell
Set-PrtgClient -LogLevel All

Get-Probe <probeName> -Verbose
```

where **&lt;probeName&gt;** is your probe name

and post the output? (please remove any sensitive details like your PRTG URL, username and passhash from the output)

**Comment by carsten0407** [comment]
Hi Lordmiko,

here the following output:

First one not functionally, installed after the PRTG-Update

Get-Probe -Name "ProbeName1" -Verbose

VERBOSE: Get-Probe: Synchronously executing request https://monitoring.de/api/table.xml?content=probenode&columns=objid,name,condition,fold,groupnum,devicenum,upsens,downsens,downacksens,partialdownsens,warnsens,pausedsens,unusualsens,undefinedsens,totalsens,schedule,basetype,baselink,notifiesx,intervalx,access,dependency,position,status,comments,priority,message,parentid,tags,type,active&count=*&filter_name= ProbeName1&filter_type=probenode&username=AA_Basic-PowershellUser&passhash=3424291689
VERBOSE: Get-Probe: <?xml version="1.0" encoding="UTF-8"?>
  <probenode totalcount="0" listend="1">
   <prtg-version>21.2.67.1562</prtg-version>
  </probenode>

----------------------------------------------------------------------------------------------------------
Second one is OK , installed before the PRTG-Update

Get-Probe -Name " ProbeName2" -Verbose

VERBOSE: Get-Probe: Synchronously executing request https://monitoring.de/api/table.xml?content=probenode&columns=objid,name,condition,fold,groupnum,devicenum,upsens,downsens,downacksens,partialdownsens,warnsens,pausedsens,unusualsens,undefinedsens,totalsens,schedule,basetype,baselink,notifiesx,intervalx,access,dependency,position,status,comments,priority,message,parentid,tags,type,active&count=*&filter_name= ProbeName2&filter_type=probenode&username=AA_Basic-PowershellUser&passhash=3424291689
VERBOSE: Get-Probe: <?xml version="1.0" encoding="UTF-8"?>
  <probenode totalcount="1" listend="1">
   <prtg-version>21.2.67.1562</prtg-version>
   <item>
    <objid>128662</objid>
    <name> ProbeName2)</name>
    <condition>Verbunden</condition>
    <condition_raw>2</condition_raw>
    <fold>False</fold>
    <fold_raw>0</fold_raw>
    <groupnum>34</groupnum>
    <groupnum_raw>0000000034</groupnum_raw>
    <devicenum>5</devicenum>
    <devicenum_raw>0000000005</devicenum_raw>
    <upsens>&lt;a title=&quot;ProbeName2: 20x OK&quot; class=&quot;sensorlinkgreen&quot; href=&quot;sensors.htm?id=128662&amp;filter_status=2&amp;filter_status=3&quot;&gt;&lt;div class=&quot;sensg&quot;&gt;20&lt;/div&gt;&lt;/a&gt;</upsens>
    <upsens_raw>0000000020</upsens_raw>
    <downsens/>
    <downsens_raw>0000000000</downsens_raw>
    <downacksens/>
    <downacksens_raw>0000000000</downacksens_raw>
    <partialdownsens/>
    <partialdownsens_raw>0000000000</partialdownsens_raw>
    <warnsens>&lt;a title=&quot; ProbeName2: 2x Warnung&quot; class=&quot;sensorlinkwarn&quot; href=&quot;sensors.htm?id=128662&amp;filter_status=4&quot;&gt;&lt;div class=&quot;sensy&quot;&gt;2&lt;/div&gt;&lt;/a&gt;</warnsens>
    <warnsens_raw>0000000002</warnsens_raw>
    <pausedsens>&lt;a title=&quot;ProbeName2: 2x Pausiert&quot; class=&quot;sensorlinkpaused&quot; href=&quot;sensors.htm?id=128662&amp;filter_status=7&amp;filter_status=8&amp;filter_status=9&amp;filter_status=11&amp;filter_status=12&quot;&gt;&lt;div class=&quot;sensb&quot;&gt;2&lt;/div&gt;&lt;/a&gt;</pausedsens>
    <pausedsens_raw>0000000002</pausedsens_raw>
    <unusualsens/>
    <unusualsens_raw>0000000000</unusualsens_raw>
    <undefinedsens/>
    <undefinedsens_raw>0000000000</undefinedsens_raw>
    <totalsens>24</totalsens>
    <totalsens_raw>0000000024</totalsens_raw>
    <schedule/>
    <basetype>probe</basetype>
    <baselink>/probenode.htm?id=128662</baselink>
    <baselink_raw>128662</baselink_raw>
    <notifiesx>Übernommen</notifiesx>
    <notifiesx_raw>Inherited</notifiesx_raw>
    <intervalx>90</intervalx>
    <intervalx_raw>0000000090</intervalx_raw>
    <access>Vollzugriff</access>
    <access_raw>0000000400</access_raw>
    <dependency>Übergeordnetes Objekt</dependency>
    <dependency_raw>Hauptgruppe</dependency_raw>
    <position>800</position>
    <position_raw>0000000800</position_raw>
    <status>OK </status>
    <status_raw>3</status_raw>
    <priority>3</priority>
    <message>&lt;div class=&quot;status&quot;&gt;OK&lt;div class=&quot;moreicon&quot;&gt;&lt;/div&gt;&lt;/div&gt;</message>
    <message_raw>OK</message_raw>
    <parentid>0</parentid>
    <tags>transitionApothekeMag</tags>
    <type>Probe</type>
    <type_raw>probenode</type_raw>
    <active>True</active>
    <active_raw>-1</active_raw>
   </item>
  </probenode>

Name                      Id     ProbeStatus  Devices                 Groups                  Up Sensors              Down                    Down (Ack)             Warning                Paused
----                      --     -----------  -------                 ------                  ----------              ----                    ----------             -------                ------
ProbeName2 128662 Connected    5                       34                      20                      0                       0                      2                      2

Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
                Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany

Phone: +49 (711) 1385 - 131

Fax:       +49 (711) 1385 - 222
Mail:      ***@***.***

Web:     www.konicaminolta.de


District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath

Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.




Von: lordmilko ***@***.***>
Gesendet: Montag, 10. Mai 2021 23:57
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)

Hi @carsten0407,
Can youu please do
Set-PrtgClient -LogLevel All

Get-Probe <probeName> -Verbose
where <probeName> is your probe name
and post the output? (please remove any sensitive details like your PRTG URL, username and passhash from the output)
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: lordmilko ***@***.***>
Gesendet: Montag, 10. Mai 2021 23:57
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Hi @carsten0407<https://github.com/carsten0407>,

Can youu please do

Set-PrtgClient -LogLevel All



Get-Probe <probeName> -Verbose

where <probeName> is your probe name

and post the output? (please remove any sensitive details like your PRTG URL, username and passhash from the output)

**Comment by lordmilko** [comment]
Hi @carsten0407,

I don't understand your response. 

First you said

> First one not functionally, installed after the PRTG-Update

But then you said

> Second one is OK , installed before the PRTG-Update

However both of these XML responses show the same PRTG version: 21.2.67.1562

When you say "PRTG-Update" do you actually mean *PrtgAPI Update*? i.e. is the top one the response given by PrtgAPI 0.9.15 while the bottom one is the response given by PrtgAPI 0.9.14? In the first response I can see you are searching for "ProbeName1" but in the second response it is "ProbeName2". I'm not sure if this is just because you modified the output to hide some sensitive info, however if these API requests are for different probes it is impossible to compare these properly.

Also, please be careful when responding to GitHub issues - not only did you leak the URL of your PRTG web server, you also appear to have leaked the username and passhash that you use for tyour API requests, in addition to all of the information contained in your mail signature when you responded to the email

**Comment by carsten0407** [comment]
Hi Lordmilko,

However both of these XML responses show the same PRTG version: 21.2.67.1562

è YES, both response are from the same PRTG Core-Server

When you say "PRTG-Update" do you actually mean PrtgAPI Update?

--> No, I mean the PRTG-Update to 21.2.67.1562 this one with the “Core Health (Autonomous)” Sensor

i.e. is the top one the response given by PrtgAPI 0.9.15 while the bottom one is the response given by PrtgAPI 0.9.14?

--> Iam allready updated to the PrtgAPI 0.9.15

In the first response I can see you are searching for "ProbeName1" but in the second response it is "ProbeName2".

è    The ProbeName1 was installed after the update to 21.2.67.1562 this one is visible with get-probe –name ProbeName1

è    The ProbeName2 was installed before the update to 21.2.67.1562 this one is visible with get-probe –name ProbeName2

however if these API requests are for different probes it is impossible to compare these properly.

è Again, I see all the probes with get-probe but not the new one installed after the PRTG-Update to 21.2.67.1562 and again, I updated to PrtgAPI 0.9.15

Also, please be careful when responding to GitHub issues - not only did you leak the URL of your PRTG web server, you also appear to have leaked the username and passhash that you use for tyour API requests, in addition to all of the information contained in your mail signature when you responded to the email

è Sorry for the signature, the PRTG output is changed

**Comment by lordmilko** [comment]
Hi @carsten0407,

I see what you're saying; I tried adding a new probe to my test server, and sure enough when I do `Get-Probe` - even without any additional arguments - the new probe server does not show up.

If I do `Get-Object -Id <id of the new probe>` it does show up, but if I then do `Get-Object -Id <id of the new probe> -Type probenode` the object isn't returned at all, despite the fact you can clearly see in the XML response the type is `probenode`

This seems like a bug in PRTG; I will speak with Paessler

**Comment by carsten0407** [comment]
Hi Lordmilko,

the same to me


Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: lordmilko ***@***.***>
Gesendet: Dienstag, 11. Mai 2021 10:54
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Hi @carsten0407<https://github.com/carsten0407>,

I see what you're saying; I tried adding a new probe to my test server, and sure enough when I do Get-Probe - even without any additional arguments - the new probe server does not show up.

If I do Get-Object -Id <id of the new probe> it does show up, but if I then do Get-Object -Id <id of the new probe> -Type probenode the object isn't returned at all, despite the fact you can clearly see in the XML response the type is probenode

This seems like a bug in PRTG; I will speak with Paessler

**Comment by carsten0407** [comment]
Hi Lordmilko,

It is strange, I found a probe, which is definitely installed before the update, and those one is also not visible with get-probe –name PROBNAME

**Comment by lordmilko** [comment]
Hi @carsten0407,

Possibly a slightly earlier PRTG update caused this behavior, however I have reached out to Paessler and they have confirmed they can reproduce this behavior as well; as such at this time this is a bug in PRTG so we will need to wait to hear back from Paessler

**Comment by carsten0407** [comment]
…yes, thanks I want you only to inform about this behaviour, this was new for me...

**Comment by lordmilko** [comment]
Hi @carsten0407,

I have pushed a new pre-release version of PrtgAPI 0.9.16 which includes a new fix for the original issue that doesn't also have side effects due to other bugs in PRTG.

Are you able to download the latest build via the [manual](https://github.com/lordmilko/PrtgAPI/wiki/Installation#manual) installation instructions and advise whether you can now retrieve all probes properly?

Regards,
lordmilko

**Comment by lordmilko** [comment]
Hi @carsten0407,

Are you able to advise whether the new build resolves your issue? I would like to release PrtgAPI 0.9.16 including this fix however would like to confirm the fix does indeed work properly first

Regards,
lordmilko

**Comment by carsten0407** [comment]
Hi lordmilko

I am in Holiday next tuesday you get an answer

Holen Sie sich Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: lordmilko ***@***.***>
Gesendet: Wednesday, May 26, 2021 11:17:44 AM
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Hi @carsten0407<https://github.com/carsten0407>,

Are you able to advise whether the new build resolves your issue? I would like to release PrtgAPI 0.9.16 including this fix however would like to confirm the fix does indeed work properly first

Regards,
lordmilko

**Comment by carsten0407** [comment]
Hi Lordmilko,

after the update it works, so the first try..
I get all the probes before the update I get nothing back…

Thanx for that

Carsten

**Comment by carsten0407** [comment]
Hi Lordmilko,

I made a mistake!

Can you tell me where I can download the PrtgAPI 0.9.16

On https://github.com/lordmilko/PrtgAPI I found only the 15er Version



Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: Trautmann, Carsten
Gesendet: Montag, 31. Mai 2021 08:09
An: lordmilko/PrtgAPI ***@***.***>; lordmilko/PrtgAPI ***@***.***>
Cc: Mention ***@***.***>
Betreff: AW: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)

Hi Lordmilko,

after the update it works, so the first try..
I get all the probes before the update I get nothing back…

Thanx for that

Carsten

**Comment by lordmilko** [comment]
PrtgAPI 0.9.16 is currently in pre-release

Please see the [manual](https://github.com/lordmilko/PrtgAPI/wiki/Installation#manual) installation instructions for details on how to download and run it

**Comment by carsten0407** [comment]
I already done this, but the version number shows

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     0.9.15     PRTGAPI                             {Add-Device, Add-Group, Add-NotificationTrigger, Add-Sensor...}

If this is correct, the new probes are not visible…

Sorry

**Comment by lordmilko** [comment]
Hi @carsten0407,

The version number is the same (since it isn't the proper 0.9.16 yet).

Are you able to provide the output of the following

```powershell
Set-PrtgClient -LogLevel All

Get-Object -Type probenode

Get-Probe -Verbose

Get-Probe <probeName> -Verbose
```

Where **<probeName>** is the name of the probe that isn't being returned

Please obscure any sensitive details such as your server name, username and passhash in the result

**Comment by carsten0407** [comment]
Get-Object -Type probenode
All the probes coming except the new installed
------------------------------------------------------------------------
Get-Probe -id  136605 –Verbose

VERBOSE: Get-Probe: Synchronously executing request https://??????????????????/api/table.xml?content=probenode&columns=objid,name,condition,fold,groupnum,devicenum,upsens,downsens,downacksens,partialdownsens,warnsens,pausedsens,unusualsens,undefinedsens,totalsens,schedule,basetype,baselink,notifiesx,intervalx,access,dependency,position,status,comments,priority,message,parentid,tags,type,active&count=*&filter_objid=136605&filter_type=probenode&username=?????????PowershellUser&passhash=???????????
VERBOSE: Get-Probe: <?xml version="1.0" encoding="UTF-8"?>
  <probenode totalcount="0" listend="1">
   <prtg-version>21.2.67.1562</prtg-version>
  </probenode>

Get-Probe -name test –Verbose

VERBOSE: Get-Probe: Synchronously executing request https://??????????????????/api/table.xml?content=probenode&columns=objid,name,condition,fold,groupnum,devicenum,upsens,downsens,downacksens,partialdownsens,warnsens,pausedsens,unusualsens,undefinedsens,totalsens,schedule,basetype,baselink,notifiesx,intervalx,access,dependency,position,status,comments,priority,message,parentid,tags,type,active&count=*&filter_name=test&filter_type=probenode&username=???????PowershellUser&passhash=????????????
VERBOSE: Get-Probe: <?xml version="1.0" encoding="UTF-8"?>
  <probenode totalcount="0" listend="1">
   <prtg-version>21.2.67.1562</prtg-version>
  </probenode>


Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: lordmilko ***@***.***>
Gesendet: Dienstag, 1. Juni 2021 10:35
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Hi @carsten0407<https://github.com/carsten0407>,

The version number is the same (since it isn't the proper 0.9.16 yet).

Are you able to provide the output of the following

Set-PrtgClient -LogLevel All



Get-Object -Type probenode



Get-Probe -Verbose



Get-Probe <probeName> -Verbose

Where is the name of the probe that isn't being returned

Please obscure any sensitive details such as your server name, username and passhash in the result

**Comment by lordmilko** [comment]
Hi @AlphaJosh,

It doesn't appear that you are using the new pre-release version of PrtgAPI

```powershell
Get-Probe -name test –Verbose

VERBOSE: Get-Probe: Synchronously executing request https://??????????????????/api/table.xml?content=probenode&columns=objid,name,condition,fold,groupnum,devicenum,upsens,downsens,downacksens,partialdownsens,warnsens,pausedsens,unusualsens,undefinedsens,totalsens,schedule,basetype,baselink,notifiesx,intervalx,access,dependency,position,status,comments,priority,message,parentid,tags,type,active&count=*&filter_name=test&filter_type=probenode&username=???????PowershellUser&passhash=????????????
```

In this API request, you can see that it executes `filter_type=probenode` - however this is the old PrtgAPI 0.9.15 behavior. If you are using the PrtgAPI 0.9.16 pre-release this should say `filter_parentid=0`

To make sure you are using the new version of PrtgAPI, can you

1. Download the [latest build](https://ci.appveyor.com/api/projects/lordmilko/prtgapi/artifacts/PrtgAPI.zip)
2. Right click the downloaded **PrtgAPI.zip** file and go to Properties
3. On the *General* tab, under *Security* select **Unblock**
4. Unzip the file to `C:\temp\PrtgAPI`
5. Open a new PowerShell and type `ipmo C:\temp\PrtgAPI`
6. Provide the output of

```powershell
Get-Probe -name test –Verbose
(gmo prtgapi).path
```

**Comment by carsten0407** [comment]
There is nothing to do for me her?
Right?

**Comment by lordmilko** [comment]
Hi @carsten0407,

Sorry, I accidentally @'d @AlphaJosh instead of you. Are you please follow the instructions above?

**Comment by carsten0407** [comment]
I did it, and it works…

THANX


Mit freundlichen Grüßen / best regards

i. A. Carsten Trautmann

Senior System Engineer
Operation Center
IT-Service
[http://support.it.konicaminolta.de/signatures/KM_LOGO.jpg]
Konica Minolta Business Solutions Deutschland GmbH
Mittlerer Pfad 1, 70499 Stuttgart, Germany
Phone:
+49 (711) 1385 - 131<tel:+497111385131>
Fax:
+49 (711) 1385 - 222
Mail:
***@***.******@***.***>
Web:
www.konicaminolta.de<http://www.konicaminolta.de>
District Court / Amtsgericht: Hannover, HRB 61209
Management / Geschäftsführer:
Joerg Hartmann & Kenichiro Fukasawa
Aufsichtsratsvorsitzender: Jens Venrath
Informationen über die Verarbeitung Ihrer personenbezogenen Daten durch uns können Sie unter https://www.konicaminolta.de/de/business-solutions/meta/dseg.html einsehen.

[http://support.it.konicaminolta.de/signatures/emailPrinzip.jpg]

Von: lordmilko ***@***.***>
Gesendet: Dienstag, 1. Juni 2021 11:07
An: lordmilko/PrtgAPI ***@***.***>
Cc: Trautmann, Carsten ***@***.***>; Mention ***@***.***>
Betreff: Re: [lordmilko/PrtgAPI] After Update to PRTG Version 21.2.67.1562 (Stable) get-probe does not give anything back (#215)


Hi @carsten0407<https://github.com/carsten0407>,

Sorry, I accidentally @'d @AlphaJosh<https://github.com/AlphaJosh> instead of you. Are you please follow the instructions above?
